File: 1985.find-the-kth-largest-integer-in-the-array.py
#
# @lc app=leetcode id=1985 lang=python3
#
# [1985] Find the Kth Largest Integer in the Array
#

# @lc code=start
from typing import List


class Solution:
    '''
    Use Z fill and sort
    '''
    def kthLargestNumber(self, nums: List[str], k: int) -> str:
        arr = []
        n = max([len(x) for x in nums])
        for i, num in enumerate(nums):
            arr.append([num.zfill(n), i])
        arr.sort(reverse=True)
        return nums[arr[k - 1][1]]
# @lc code=end

File: 1985.find-the-kth-largest-integer-in-the-array.cs
/*
 * @lc app=leetcode id=1985 lang=csharp
 *
 * [1985] Find the Kth Largest Integer in the Array
 */

// @lc code=start
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public string KthLargestNumber(string[] nums, int k) {
        List<Tuple<string, int>> arr = new List<Tuple<string, int>>();
        int n = nums.Select(x=>x.Length).Max();

        foreach (var num in nums) {
            arr.Add(new Tuple<string, int>(num.PadLeft(n, '0'), arr.Count));
        }

        arr.Sort((a, b) => string.Compare(b.Item1, a.Item1, StringComparison.Ordinal));

        return nums[arr[k - 1].Item2];
    }
}

// @lc code=end

